lib/diagnostic.js

```diff
@@ -16,8 +16,23 @@
 const fs = require('fs'); // jshint ignore: line
 
 const sumLines = (filename, callback) => {
-  /* your response here */
-  callback();
+  let sum = 0;
+  fs.readFile(filename, 'utf8', (error, data) => {
+    if (error) {
+      console.error(error.stack);
+      return;
+    }
+
+  let dataArray = data.split('\n');
+    for (let i = 0; i < dataArray.length; i++) {
+      let line = parseInt(dataArray[i]);
+      if (typeof line !== 'number') {
+        callback(new Error('line not a number'));
+      }
+      sum += dataArray;
+    }
+  });
+  callback(null, sum);
 };
```

Comment:
When I tried console.logging `sum` when running this with integers.txt, I got this: `sum is  00,1,1,2,3,5,8,13,21,34,`

This is happening because of line 32. Remember, Javascript has no sense of the difference between strings and integers. Thus, when you append the `dataArray` to it, it's happy to make it a longer string!